### ISSUE TITLE
Site editor: Safari exposes the preview iframe as 'ignored' element

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59319]
### Description

While working on https://github.com/WordPress/gutenberg/pull/59317 I noticed something that seems specific to Safari.

When the editor is in 'view' mode, the `role` of the preview iframe is set to `button`. That's because we want to make the site preview clickable to switch the editor to edit mode and we want to provide that functionality also to keyboard users.

However, Safari exposes the iframe as 'ignored' element. To test:
- Use Safari.
- Go to the Site editor in view mode.
- In the safari dev tools > Inspector, select the iframe then go to the Node tab, scroll the tab content to the bottom and observe the 'Accessibility' panel. Screenshot:

![Screenshot 2024-02-23 at 16 25 13](https://github.com/WordPress/gutenberg/assets/1682452/07416787-5351-488d-944e-86e9da160792)

- Observe the panel reports 'Ignored Yes', although the iframe has a role="button" and an aria-label="Editor canvas"..

What that does mean for screen reader users?

When using Safari and VoiceOver, the iframe is not exposed in the accessibility tree. As such, VoiceOver can't announce it's a button and can't announce its label. Safari and VoiceOver users will have no clue where focus is and no clue about the available action.

To test:
- Use Safari and VoiceOver.
- Go to the Site editor in view mode.
- Tab through the focusable elements until you reach the 'Saved' button at the bottom of the navigation panel.
- Tab again one or two times.
- Observe that while there are tab stops, no button with name 'Editor canvas' is ever announced.
- Typically, VoiceOver will announce the Editor content region and report it is empty, because all elements within the iframe use `inert` and `aria-hidden="true"`.
- However, `inert` and `aria-hidden="true"` are unrelated to the missed announcement of the iframe.
- Theoretically the iframe should be announced as a button named 'Editor canvas' but Safari just ignores it.

Screenshot:

![Screenshot 2024-02-23 at 16 12 04](https://github.com/WordPress/gutenberg/assets/1682452/7eba94c7-fbe9-40ae-ad35-67d5745961d8)

Overall, overriding the native role of the iframe seems a bit hacky and it appears it is not expected by some browsers. I'd think the only solution to this would be using a real button element overlaid on top of the preview iframe. If i recall correctly, a previous implementation did use a button element and that was changed at some point.

### Step-by-step reproduction instructions

See instructions above

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
> However, inert and aria-hidden="true" are unrelated to the missed announcement of the iframe.

After some more testing, I'm not sure that's the case. I think it's likely a mix of the two things.
